docs/program.rst

```diff
@@ -1,4 +1,41 @@
 Program
 =======
 
-contains Program details
\ No newline at end of file
+In the OpenQL programming model, one first creates the platform object and then with it the program object.
+After that, one creates kernels with gates and adds these kernels to the program.
+Finally, one compiles the program and executes it.
+At any time, options can be set and got.
+
+Platform creation takes a name (to use in information messages) and the name of the platform configuration file.
+The latter is used to initialize the platform attributes, e.g. to create custom gates.
+
+A program is created by specifying a name, the platform, and the numbers of quantum and classical registers.
+The latter defaults to 0.
```

Comment:
`0` as an integer, as a string, as a `None` value?  
What does this default mean? 
Might be useful to link to the API autodoc as that is usally more clear than explaining that here.

Reply:
I deleted the remark that the number of classical registers has a default 0 when creating a program. Should be in the API definition indeed.